File: src/nl/rubensten/texifyidea/structure/latex/LatexFilePresentation.java
package nl.rubensten.texifyidea.structure.latex;

import com.intellij.navigation.ItemPresentation;
import com.intellij.psi.PsiFile;
import nl.rubensten.texifyidea.TexifyIcons;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.util.regex.Pattern;

/**
 * @author Ruben Schellekens
 */
public class LatexFilePresentation implements ItemPresentation {

    private static final Pattern EXTENSION = Pattern.compile("\\.[a-zA-Z0-9]+$");

    private final PsiFile file;
    private final String presentableText;

    public LatexFilePresentation(PsiFile file) {
        this.file = file;
        this.presentableText = EXTENSION.matcher(file.getName()).replaceAll("");
    }

    @Nullable
    @Override
    public String getPresentableText() {
        return presentableText;
    }

    @Nullable
    @Override
    public String getLocationString() {
        return file.getVirtualFile().getPath();
    }

    @Nullable
    @Override
    public Icon getIcon(boolean b) {
        return TexifyIcons.DOT_LATEX;
    }
}

File: src/nl/rubensten/texifyidea/structure/latex/LatexFilePresentation.kt
package nl.rubensten.texifyidea.structure.latex

import com.intellij.navigation.ItemPresentation
import com.intellij.psi.PsiFile
import nl.rubensten.texifyidea.TexifyIcons
import nl.rubensten.texifyidea.util.Magic

/**
 * @author Ruben Schellekens
 */
open class LatexFilePresentation(private val file: PsiFile) : ItemPresentation {

    private val presentableText = Magic.Pattern.fileExtension.matcher(file.name).replaceAll("")

    override fun getPresentableText() = presentableText!!

    override fun getLocationString() = file.virtualFile.path

    override fun getIcon(b: Boolean) = TexifyIcons.DOT_LATEX!!
}